plugins/scaffolder/src/extensions/types.ts

```diff
@@ -29,3 +29,10 @@ export type FieldExtensionOptions<T = any> = {
   component: (props: FieldProps<T>) => JSX.Element | null;
   validation?: CustomFieldValidator<T>;
 };
+
+export interface CustomFieldExtension<ReturnValue, UiOptions extends {} = {}>
```

Comment:
To not forget the discussion: thinking `FieldExtensionProps` or `FieldExtensionComponentProps` makes sense here to match the `FieldExtensionOptions` above